lib/toy_robot_elixir/actions/move.ex

```diff
@@ -0,0 +1,33 @@
+defmodule ToyRobotElixir.Actions.Move do
+  @behaviour ToyRobotElixir.Actions.Action
+
+  alias ToyRobotElixir.{Robot, Table}
+
+  @impl true
+  def execute(robot, _) do
+    %Robot{robot | x: new_x(robot), y: new_y(robot)}
+  end
+
+  @impl true
+  def validate(nil, _, []), do: {:error, "Robot not placed"}
+
+  def validate(robot, %Table{} = table, []) do
+    if Table.is_out_of_table?(new_x(robot), new_y(robot), table) do
```

Comment:
Uses table to validate if it is out of bounds, table is standard 5x5 bit it can be any size, the action don't care about it